packages/cheqd/tests/helpers/cheqdAnonCredsSetup.ts

```diff
@@ -0,0 +1,356 @@
+} = {}) => {
+  const dataIntegrityCredentialFormatService = new DataIntegrityCredentialFormatService()
+  const anonCredsCredentialFormatService = new AnonCredsCredentialFormatService()
+  const anonCredsProofFormatService = new AnonCredsProofFormatService()
+  const presentationExchangeProofFormatService = new PresentationExchangeProofFormatService()
+
+  const modules = {
+    cheqdSdk: new CheqdModule(getCheqdModuleConfig(seed, rpcUrl)),
+    credentials: new CredentialsModule({
+      autoAcceptCredentials,
+      credentialProtocols: [
+        new V2CredentialProtocol({
+          credentialFormats: [dataIntegrityCredentialFormatService, anonCredsCredentialFormatService],
+        }),
+      ],
+    }),
+    proofs: new ProofsModule({
+      autoAcceptProofs,
+      proofProtocols: [
+        new V2ProofProtocol({
+          proofFormats: [anonCredsProofFormatService, presentationExchangeProofFormatService],
+        }),
+      ],
+    }),
+    anoncreds: new AnonCredsModule({
+      registries: registries ?? [new InMemoryAnonCredsRegistry()],
+      anoncreds,
+    }),
+    dids: new DidsModule({
+      registrars: [new CheqdDidRegistrar()],
+      resolvers: [new CheqdDidResolver()],
+    }),
+    cache: new CacheModule({
+      cache: new InMemoryLruCache({ limit: 100 }),
+    }),
+  } as const
+
+  return modules
+}
+
+interface SetupAnonCredsTestsReturn<VerifierName extends string | undefined, CreateConnections extends boolean> {
+  issuerAgent: AnonCredsTestsAgent
+  issuerReplay: EventReplaySubject
+
+  holderAgent: AnonCredsTestsAgent
+  holderReplay: EventReplaySubject
+
+  issuerHolderConnectionId: CreateConnections extends true ? string : undefined
+  holderIssuerConnectionId: CreateConnections extends true ? string : undefined
+
+  issuerId: string
+
+  verifierHolderConnectionId: CreateConnections extends true
+    ? VerifierName extends string
+      ? string
+      : undefined
+    : undefined
+  holderVerifierConnectionId: CreateConnections extends true
+    ? VerifierName extends string
+      ? string
+      : undefined
+    : undefined
+
+  verifierAgent: VerifierName extends string ? AnonCredsTestsAgent : undefined
+  verifierReplay: VerifierName extends string ? EventReplaySubject : undefined
+
+  schemaId: string
+  credentialDefinitionId: string
+}
+
+async function createDid(agent: AnonCredsTestsAgent) {
+  const privateKey = TypedArrayEncoder.fromString('000000000000000000000000000cheqd')
+  const did = await agent.dids.create<CheqdDidCreateOptions>({
+    method: 'cheqd',
+    secret: {
+      verificationMethod: {
+        id: 'key-10',
+        type: 'Ed25519VerificationKey2020',
+        privateKey,
+      },
+    },
+    options: {
+      network: 'testnet',
+      methodSpecificIdAlgo: 'uuid',
+    },
+  })
+  expect(did.didState).toMatchObject({ state: 'finished' })
+  return did.didState.did as string
+}
+
+export async function setupAnonCredsTests<
```

Comment:
This is a lot of duplication for swapping out cheqd vs indy-vdr. Would it be easier to combine them and add some conditional steps?  Basically everything in this file is duplicated except for the cheqd specific stuff (which is very minimal)